app/validators/alphabetic_validator.rb

```diff
@@ -0,0 +1,20 @@
+# AlphabeticValidator validates that there's only alphabetic digits in a set of
+# protected values.
+class AlphabeticValidator < ActiveModel::EachValidator
+  PROTECTED_VALUES = [:company_name, :company_unit, :country, :state, :city].freeze
+
+  def validate_each(record, attribute, value)
+    return unless protected?(record.pillar)
+    return if value.nil? || value.match(/^[a-zA-Z\s]+$/)
+
+    record.errors[attribute] << "contains invalid characters"
+  end
+
+  protected
+
+  # Returns true if the given value belongs to a protected key.
+  def protected?(pillar)
+    keys = Pillar.all_pillars
+    PROTECTED_VALUES.any? { |v| keys[v] == pillar }
```

Comment:
Not really. That's what I tried at first, but bear in mind that the `PROTECTED_VALUES` constant only contain the key for which we identify the different pillar keys, which are the ones to be compared then after. A bit confusing, but we cannot simplify it :P